File: backend/internal/domain/course.go
package domain

import (
	"database/sql"
	"encoding/json"
	"time"
)

type NullInt64 struct {
	sql.NullInt64
}

func (n NullInt64) MarshalJSON() ([]byte, error) {
	if !n.Valid {
		return json.Marshal(nil)
	}
	return json.Marshal(n.Int64)
}

type Course struct {
	ID          int            `json:"id"`
	Code        string         `json:"code"`
	Name        string         `json:"name"`
	SKS         int            `json:"sks"`
	Description string         `json:"description"`
	CourseType  string         `json:"course_type"`
	CawuID      NullInt64      `json:"cawu_id"`
	Instructors []string       `json:"instructors"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
}

type CourseRepository interface {
	Create(course *Course) error
	GetAll(page, limit int) ([]Course, int, error)
	GetByID(id int) (*Course, error)
	Update(course *Course) error
	Delete(id int) error
}

File: frontend/src/types/course.ts
export interface Course {
  id: number;
  code: string;
  name: string;
  sks: number;
  description?: string;
  instructors?: string[];
  cawu_id?: number;
  course_type?: "lecturer" | "lab";
  slug: string;
  created_at: string;
  updated_at: string;
}

export interface CourseInput {
  code: string;
  name: string;
  sks: number;
  description?: string;
  instructors?: string[];
  cawu_id?: number;
  course_type?: "lecturer" | "lab";
}
